agent/engine/engine_integ_test.go

```diff
@@ -125,6 +127,26 @@ func verifyTaskStoppedStateChange(t *testing.T, taskEngine TaskEngine) {
 		"Expected task to be STOPPED")
 }
 
+func dialWithRetries(proto string, address string, tries int, timeout time.Duration) (net.Conn, error) {
```

Comment:
why not use the `utils.SimpleBackoff()` instead?

Reply:
I don't think there is a particular reason to not, these codes are all inherited from the existing integration test code. I will try to refactor it a little bit. The same for the other comments.